### ISSUE TITLE
SnakemakeSL Assertion failed: Text of a single string literal node expected: "" 

### ISSUE THREAD

**Issue by nekit-000000** [JetBrains-Research/snakecharm#190]
```
java.lang.Throwable: Assertion failed: Text of a single string literal node expected: "" 
    at com.intellij.openapi.diagnostic.Logger.assertTrue(Logger.java:180)
    at com.jetbrains.python.psi.PyStringLiteralUtil.getContentRange(PyStringLiteralUtil.java:128)
    at com.jetbrains.python.psi.impl.PyPlainStringElementImpl.getContentRange(PyPlainStringElementImpl.java:48)
    at com.jetbrains.python.psi.impl.PyStringLiteralDecoder.decodeContent(PyStringLiteralDecoder.java:70)
    at com.jetbrains.python.psi.impl.PyPlainStringElementImpl.getDecodedFragments(PyPlainStringElementImpl.java:42)
    at com.jetbrains.python.psi.impl.PyStringLiteralExpressionImpl.lambda$getDecodedFragments$2(PyStringLiteralExpressionImpl.java:102)
    at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:271)
    at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1654)
    at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
    at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
    at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:550)
    at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
    at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:517)
    at one.util.streamex.AbstractStreamEx.toArray(AbstractStreamEx.java:344)
    at one.util.streamex.AbstractStreamEx.toList(AbstractStreamEx.java:1186)
    at com.jetbrains.python.psi.impl.PyStringLiteralExpressionImpl.getDecodedFragments(PyStringLiteralExpressionImpl.java:107)
    at com.jetbrains.python.psi.impl.PyStringLiteralExpressionImpl.getStringValue(PyStringLiteralExpressionImpl.java:143)
    at com.jetbrains.snakecharm.string_language.SmkSLInjector.containsBraces(SmkSLInjector.kt:38)
    at com.jetbrains.snakecharm.string_language.SmkSLInjector.isValidForInjection(SmkSLInjector.kt:59)
    at com.jetbrains.snakecharm.string_language.SmkSLInjector.registerInjection(SmkSLInjector.kt:32)
    at com.jetbrains.python.codeInsight.PyInjectorBase.getLanguagesToInject(PyInjectorBase.java:33)
    at com.intellij.psi.impl.source.tree.injected.InjectedLanguageManagerImpl.processInPlaceInjectorsFor(InjectedLanguageManagerImpl.java:421)
    at com.intellij.psi.impl.source.tree.injected.InjectedLanguageUtil.probeElementsUp(InjectedLanguageUtil.java:348)
    at com.intellij.psi.impl.source.tree.injected.InjectedLanguageUtil.enumerate(InjectedLanguageUtil.java:173)
    at com.intellij.psi.impl.source.tree.injected.InjectedLanguageUtil.findInside(InjectedLanguageUtil.java:490)
    at com.intellij.psi.impl.source.tree.injected.InjectedLanguageUtil.tryOffset(InjectedLanguageUtil.java:469)
    at com.intellij.psi.impl.source.tree.injected.InjectedLanguageUtil.findInjectedElementNoCommit(InjectedLanguageUtil.java:449)
    at com.intellij.psi.impl.source.tree.injected.InjectedLanguageManagerImpl.findInjectedElementAt(InjectedLanguageManagerImpl.java:286)
    at com.intellij.psi.impl.source.tree.injected.InjectedLanguageUtil.findInjectedPsiNoCommit(InjectedLanguageUtil.java:303)
    at com.intellij.codeInsight.intention.impl.ShowIntentionActionsHandler.chooseBetweenHostAndInjected(ShowIntentionActionsHandler.java:151)
    at com.intellij.codeInsight.daemon.impl.ShowIntentionsPass.getActionsToShow(ShowIntentionsPass.java:297)
    at com.intellij.codeInsight.daemon.impl.ShowIntentionsPass.doCollectInformation(ShowIntentionsPass.java:220)
    at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:55)
    at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$null$1(PassExecutorService.java:429)
    at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1106)
    at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:422)
    at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:591)
    at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:537)
    at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:59)
    at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:421)
    at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:397)
    at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:164)
    at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:204)
    at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:395)
    at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:161)
    at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
    at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
    at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
    at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
    at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
```

**Comment by iromeo** [comment]
Could you remind is it our bug or PyCharm?

**Comment by iromeo** [comment]
```
java.lang.Throwable: Assertion failed: Text of a single string literal node expected: "--buffer_size 20G" 
	at com.intellij.openapi.diagnostic.Logger.assertTrue(Logger.java:180)
	at com.jetbrains.python.psi.PyStringLiteralUtil.getContentRange(PyStringLiteralUtil.java:128)
	at com.jetbrains.python.psi.impl.PyPlainStringElementImpl.getContentRange(PyPlainStringElementImpl.java:48)
	at com.jetbrains.python.psi.impl.PyPlainStringElementImpl.getContent(PyPlainStringElementImpl.java:36)
	at com.jetbrains.snakecharm.stringLanguage.lang.SmkSLInjector.containsLBraceOrIsEmpty(SmkSLInjector.kt:33)
	at com.jetbrains.snakecharm.stringLanguage.lang.SmkSLInjector.isValidForInjection(SmkSLInjector.kt:65)
	at com.jetbrains.snakecharm.stringLanguage.lang.SmkSLInjector.registerInjection(SmkSLInjector.kt:26)
	at com.jetbrains.python.codeInsight.PyInjectorBase.getLanguagesToInject(PyInjectorBase.java:33)
	at com.intellij.psi.impl.source.tree.injected.InjectedLanguageManagerImpl.processInPlaceInjectorsFor(InjectedLanguageManagerImpl.java:421)
	at com.intellij.psi.impl.source.tree.injected.InjectedLanguageUtil.probeElementsUp(InjectedLanguageUtil.java:348)
	at com.intellij.psi.impl.source.tree.injected.InjectedLanguageUtil.enumerate(InjectedLanguageUtil.java:173)
	at com.intellij.psi.impl.source.tree.injected.InjectedLanguageManagerImpl.enumerateEx(InjectedLanguageManagerImpl.java:315)
	at com.intellij.codeInsight.daemon.impl.InjectedGeneralHighlightingPass.lambda$getInjectedPsiFiles$3(InjectedGeneralHighlightingPass.java:189)
	at com.intellij.concurrency.ApplierCompleter.execAndForkSubTasks(ApplierCompleter.java:133)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1106)
	at com.intellij.concurrency.ApplierCompleter.lambda$wrapInReadActionAndIndicator$1(ApplierCompleter.java:105)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:591)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:537)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:59)
	at com.intellij.concurrency.ApplierCompleter.wrapInReadActionAndIndicator(ApplierCompleter.java:116)
	at com.intellij.concurrency.ApplierCompleter.lambda$compute$0(ApplierCompleter.java:96)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:164)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:204)
	at com.intellij.concurrency.ApplierCompleter.compute(ApplierCompleter.java:96)
	at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:746)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
```
When changing
```
    params:
        extra="--buffer_size 20G"  # --ample_memory fails frequently with 'bus error (core dumped)'. In general is ~30 minutes faster,
        # but requires more RAM
```
to 
```
    params:
        extra="--buffer_size 20G"
        # --ample_memory fails frequently with 'bus error (core dumped)'. In general is ~30 minutes faster,
        # but requires more RAM
```

**Comment by iromeo** [comment]
Fixed in ec3c02e05c201693b9c2585f7ccb9e29aa881845
